File: fixtures/cpp/src/operations.cpp
#include "operations.h"

namespace operations {

double add(double a, double b) { return a + b; }

double subtract(double a, double b) { return a - b; }

double multiply(double a, double b) { return a * b; }

double divide(double a, double b) {
    if (b == 0.0) {
        throw DivisionByZeroError();
    }
    return a / b;
}

DivisionByZeroError::DivisionByZeroError() : std::runtime_error("division by zero") {}

} // namespace operations

File: fixtures/ruby/lib/operations.rb
# frozen_string_literal: true

require_relative "errors"

# Pure arithmetic operations.
module Operations
  module_function

  def add(a, b)
    a + b
  end

  def subtract(a, b)
    a - b
  end

  def multiply(a, b)
    a * b
  end

  def divide(a, b)
    raise DivisionByZeroError if b.zero?

    a.to_f / b
  end
end
